library/Haste/Image/Image.php

```diff
@@ -202,4 +202,252 @@ public static function addWatermark($image, $watermark, $position = 'br', $targe
+        }
+
+        $fileModels = \FilesModel::findMultipleByUuids($files);
+
+        if (null === $fileModels) {
+            return [];
+        }
+
+        foreach ($fileModels as $fileModel) {
+            // Single files
+            if ('file' === $fileModel->type) {
+                $file = new \File($fileModel->path, true);
+
+                if (!$file->exists() || !$file->isImage) {
+                    continue;
+                }
+
+                $images[$fileModel->path] = static::prepareImage($fileModel, $options);
+            }
+            // Folders
+            else {
+                $subFileModels = \FilesModel::findByPid($fileModel->uuid);
+
+                if (null === $subFileModels) {
+                    continue;
+                }
+
+                foreach ($subFileModels as $subFileModel) {
+                    if ('file' === $subFileModel->type) {
+                        $file = new \File($subFileModel->path, true);
+
+                        if (!$file->exists() || !$file->isImage) {
+                            continue;
+                        }
+
+                        $images[$subFileModel->path] = static::prepareImage($subFileModel, $options);
+                    }
+                }
+            }
+        }
+
+        return $images;
+    }
+
+    /**
+     * Prepares one image for a typical Contao template.
+     *
+     * Possible option keys:
+     *
+     *  - language (language for meta data. If not specified, the language of the current page object is taken)
+     *  - size (either integer with responsive image configuration ID or an array of three keys whereas 0 = width, 1 = height, 2 = crop mode)
+     *  - fullsize (if provided, adds full size handling)
+     *
+     * @param \FilesModel $fileModel
+     * @param array $options
+     *
+     * return array
+     */
+    public static function prepareImage(\FilesModel $fileModel, array $options)
+    {
+        $file = new \File($fileModel->path, true);
+
+        if (!isset($options['language'])) {
+            global $objPage;
+            $options['language'] = $objPage->language;
+        }
+
+        $meta = \Frontend::getMetaData($fileModel->meta, $options['language']);
+
+        // Use the file name as title if none is given
+        if ('' === $meta['title']) {
+            $meta['title'] = specialchars($file->basename);
+        }
+
+        $image = [
+            'id'        => $fileModel->id,
+            'uuid'      => $fileModel->uuid,
+            'name'      => $file->basename,
+            'singleSRC' => $fileModel->path,
+            'alt'       => $meta['title'],
+            'imageUrl'  => $meta['link'],
+            'caption'   => $meta['caption'],
+            'mtime'     => $file->mtime
+        ];
+
+        foreach (['size', 'fullsize'] as $k) {
+            if (isset($options[$k])) {
+                $image[$k] = $options[$k];
```

Comment:
Wouldn't it be better if I can change other things like `caption`? Say `$image = array_merge((array) $options['image'], ['id' => $fileModel->id, ...]);`